src/main/kotlin/cacophony/controlflow/FunctionHandler.kt

```diff
@@ -1,37 +1,118 @@
 package cacophony.controlflow
 
+import cacophony.controlflow.CFGNode.Addition
+import cacophony.controlflow.CFGNode.Constant
+import cacophony.controlflow.CFGNode.MemoryAccess
+import cacophony.controlflow.CFGNode.VariableUse
 import cacophony.semantic.AnalyzedFunction
 import cacophony.semantic.syntaxtree.Definition
+import cacophony.semantic.syntaxtree.Definition.FunctionDeclaration
+import cacophony.semantic.syntaxtree.Literal
+import cacophony.utils.CompileException
+import cacophony.utils.Location
 
-class FunctionHandler(
-    val function: Definition.FunctionDeclaration,
-    val analyzedFunction: AnalyzedFunction,
-) {
-    sealed class VariableAllocation() {
-        data class Reg(val register: Register)
+sealed class VariableAllocation {
+    data class Reg(
+        val register: Register,
+    ) : VariableAllocation()
 
-        data class Stack(val offset: Int)
-    }
+    data class Stack(
+        val offset: Int,
+    ) : VariableAllocation()
+}
+
+interface FunctionHandler {
```

Comment:
Why would we change that to an interface? Do we expept another implementation or are there any problems with mocking for the tests?

Reply:
yes, its for tests
can you easily mock non-interfaces in kotlin?